addon/test-waiter.ts

```diff
@@ -0,0 +1,116 @@
+import { IWaiter, WaiterName, ITestWaiterDebugInfo } from './types';
+import { register } from './waiter-manager';
+
+/**
+ * A class providing creation, registration and async waiting functionality.
```

Comment:
I hate hate hate hate hate this style of comment 😩

Reply:
It's the default when I add a jsdoc in vscode. I love love love love it!

Reply:
I'll fix them all when you are done ;)

Reply:
:)